### PR TITLE
Add CMake build system

### PR DESCRIPTION
**Pull request by GunGraveKoga**
CMake 3.15.0+ support Swift targets
On Windows CMake the only way to build a Swift project for now

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+if(CMAKE_SYSTEM_NAME STREQUAL Windows OR CMAKE_SYSTEM_NAME STREQUAL Darwin)
+  option(BUILD_SHARED_LIBS "Build shared libraries by default" YES)
+endif()
+
+add_library(XMLCoder 
+    Sources/XMLCoder/Auxiliaries/Box/BoolBox.swift
```
Comment (MaxDesiatov) [inline CMakeLists.txt]: Would there be an easy way to recursively add all `.swift` fies in `Sources/XMLCoder` directory and its subdirectories? I'm a bit afraid of people forgetting to update this list after adding a file or two to the project.

Comment (GunGraveKoga) [reply CMakeLists.txt]: OK. I will add the ability to recursively capture source files via GLOB_RECURSE

Comment (GunGraveKoga) [reply CMakeLists.txt]: Fixed

**Comment by MaxDesiatov** [comment]
@GunGraveKoga do I understand correctly that adding a CI job to test a Windows build on Azure Pipelines is not viable at the moment? Is this blocked by any of the Windows changes not merged to Swift `master` yet or something else?

**Comment by GunGraveKoga** [comment]
@MaxDesiatov as I know, main problem is a Developer Tools, which would work right out of box, like macOS and Linux Toolchains, which includes Swift toolchain and SDK (Foundation/Dispatch .etc)
I download Windows toolchain (Swift compiler and SDK) build artifacts manualy, but You can try to workaround this with PowerShell script https://github.com/compnerd/windows-swift/blob/master/utilities/Download-SwiftArtifacts.ps1
But I didn't use it else...
Any way CMake build system should work properly for macOS and Linux too

**Comment by GunGraveKoga** [comment]
You could check https://dev.azure.com/compnerd/windows-swift for more information and build status. As I know there is no an other public build-bot for Swift with available build artifacts.
I am not good with CI-things, sorry

**Comment by MaxDesiatov** [comment]
As far as I know, latest development snapshots of Swift for Windows do support SwiftPM. I'm happy to accept a PR that sets up a CI workflow for that on GitHub Actions, but I'll have to close this one unfortunately.

**Pull request closed without merging**